horace_core/sqw/page_operations/PageOp_join_sqw.m

```diff
@@ -0,0 +1,168 @@
+        block_idx_;
+        % the array of positions each combined page occupies in target
+        % dataset. Not very useful in serial mode but may be necessary in
+        % parallel mode if parallel_write is available.
+        page_start_pos_;
+        % array of current positions of all contributing pixel datasets
+        current_page_pix_pos_;
+    end
+    methods
+        function obj = PageOp_join_sqw(varargin)
+            obj = obj@PageOpBase(varargin{:});
+            obj.op_name_ = 'join_sqw';
+            obj.split_at_bin_edges = true;
+        end
+
+        function [obj,in_sqw] = init(obj,in_sqw,new_runid)
+            % initialize join_sqw algorithm.
+            % Input:
+            % in_sqw         -- special sqw object to join, prepared by
+            %                   collect_sqw_metadata algorithm.
+            if ~isa(in_sqw.pix,'MultipixBase')
+                error('HORACE:PageOp_join_sqw:invalid_argument', ...
+                    'Input sqw object does not contain information on how to combine input data')
+            end            
+            % Transfer input MultipxBase object as source of data in the
+            % operation
+            obj.pix_combine_info = in_sqw.pix;
+            in_sqw.pix = PixelDataMemory();
+            %
+            obj = init@PageOpBase(obj,in_sqw);
+            obj.new_runid = new_runid;
+            % clear signal accumulator to save memory; it will not be used
+            % here.
+            obj.sig_acc_  = [];
+            % initialize input datasets for read access
+            obj.pix_combine_info  = obj.pix_combine_info.init_pix_access();
+            obj.current_page_pix_pos_ = ones(1,obj.pix_combine_info.nfiles);
+        end
+        function [npix_chunks, npix_idx,obj] = split_into_pages(obj,npix,chunk_size)
+            % Overload of split method allowing to define large target chink
+            % and store npix_idx for internal usage
+            % Inputs:
+            % npix  -- image npix array, which defines the number of pixels
+            %           contributing into each image bin and the pixels
+            %           ordering in the linear array
+            % chunk_size
+            %       -- sized of chunks to split pixels
+            % Returns:
+            % npix_chunks -- cellarray, containing the npix parts
+            % npix_idx    -- [2,n_chunks] array of indices of the chunks in
+            %                the npix array.
+            % See split procedure for more details
+            fb = config_store.instance().get_value( ...
+                'hor_config','fb_scale_factor');
+            % do large chunk to decrease number of sub-calls to each data
+            % pixels
+            large_chunk = chunk_size*fb;
+            [npix_chunks, npix_idx,obj] = split_into_pages@PageOpBase(obj,npix,large_chunk);
+            obj.block_idx_ = npix_idx;
+            page_sizes = cellfun(@(x)sum(x(:)),npix_chunks);
```

Comment:
Could just be
```suggestion
            page_sizes = cellfun(@sum, npix_chunks);
```

Reply:
are you sure chunks have correct shape (1D) ? . Better to be safe then sorry.